src/hooks/useAsyncStorage.ts

```diff
@@ -12,12 +12,14 @@ export function useAsyncStorage(key: string) {
 
   const setItem = useCallback(
     (value: string) => {
-      if (value !== itemResult.data) {
-        AsyncStorage.setItem(key, value)
-          .catch((error) => console.error(`[LifeOmicSDK]:${error}`))
-          .finally(() => {
```

Comment:
This was causing an undefined function error that is resolved by using a try/catch here. I checked all the values at the time the error occurs, like AsyncStorage.SetItem but it all appears to be populated with functions as expected.

Reply:
I will also add that I've run into this error unrelated to this PR.